Updating the sections on unit testing (#632)
MicrosoftDocs/quantum-docs-pr: articles/techniques/testing-and-debugging.md

#### [Command Line / Visual Studio Code](#tab/tabid-vscode)

The pass/fail status for each test is printed to the console by `dotnet test`.
For failing tests, the outputs are also printed to the console to help diagnose the failure.

***

## Assertions

> One important consequence of the fact that functions in Q# have no side effects is that any effects of executing a function whose output type is the empty tuple `()` can never be observed from within a Q# program.

Reviewer: ```suggestion
Because functions in Q# have no _logical_ side effects, any _other kinds_ of effects of executing a function whose output type is the empty tuple `()` can never be observed from within a Q# program.
```
